components/pages/dashboard/ads/AdsForm.tsx

```diff
@@ -0,0 +1,319 @@
+import clsx from 'clsx'
+import { Icon } from '@iconify/react'
+import { useRouter } from 'next/router'
+import { useForm } from 'react-hook-form'
+import { button } from '~components/Dashboard/Button'
+import Field from '~components/Dashboard/Form/Field'
+import { FileInput, Input } from '~components/Dashboard/Input'
+import { RadioGroup } from '~components/Dashboard/Radio'
+import Image from 'next/image'
+import { isAfter } from 'date-fns'
+
+type FormValue = {
+  adName: string
+  startDate: string
+  endDate: string
+  image: any
+  headline: string
+  description: string
+  landingUrl: string
+  callToActionUrl: string
+  discordPlacement: string
+  telegramPlacement: string
+  metaversePlacement: string
+}
+
+export function AdsForm() {
+  const { back } = useRouter()
+  const { handleSubmit, control, watch } = useForm<FormValue>({
+    defaultValues: {
+      discordPlacement: '',
+    },
+  })
+
+  const startDate = watch('startDate')
+
+  const onSubmit = (values: FormValue) => {
+    console.log(values)
+  }
+
+  const discordOptions = [
+    {
+      src: '/assets/ads/discord-1546x423.png',
+      label: '1546 x 423',
+      value: '1',
+    },
+    {
+      src: '/assets/ads/512x512.png',
+      label: '512 x 512',
+      value: '2',
+    },
+    {
+      src: '/assets/ads/no-img.png',
+      label: 'No image',
+      value: '3',
+    },
+  ]
+
+  const telegramOptions = [
+    {
+      src: '/assets/ads/tele-1546x423.png',
+      label: '1546 x 423',
+      value: '1',
+    },
+    {
+      src: '/assets/ads/512x512.png',
+      label: '512 x 512',
+      value: '2',
+    },
+    {
+      src: '/assets/ads/no-img.png',
+      label: 'No image',
+      value: '3',
+    },
+  ]
+
+  const metaverseOptions = [
+    {
+      src: '/assets/ads/big-ad.png',
+      label: '1546 x 423',
+      value: '1',
+    },
+    {
+      src: '/assets/ads/small-ad.png',
+      label: '512 x 512',
+      value: '2',
+    },
+    {
+      src: '',
+      label: '',
+      value: '',
+    },
+  ]
+
+  const radioOptionClassName =
+    'flex flex-col w-fit p-[10px] gap-y-[10px] rounded-lg border-solid border-[3px] border-transparent cursor-pointer'
+
+  const renderCustomRadioGroup = ({
+    options,
+    optionClassName,
+  }: {
+    options: Record<string, any>[]
+    optionClassName: string
+  }) => {
+    return (
+      <RadioGroup
+        options={options.map(({ label, value }) => ({
+          label,
+          value,
+        }))}
+        radioGroupClassName="flex flex-row justify-between"
+        renderOption={(option, selectedValue) => {
+          return (
+            <div
+              className={clsx(radioOptionClassName, {
+                ' !border-mochi':
+                  selectedValue && option.value === selectedValue,
+              })}
```

Comment:
don't overuse `important` syntax
```suggestion
              className={clsx("base class", {
                'border-mochi & whatever active class':
                  selectedValue && option.value === selectedValue,
                'inactive class': !selectedValue || option.value !== selectedValue
              })}
```

Reply:
thanks for pointing this out, really appriciate it 🎉